mmengine/config/config.py

```diff
@@ -1432,7 +1435,8 @@ def _format_dict(input_dict, outest_level=False):
             use_mapping = _contain_invalid_identifier(input_dict)
             if use_mapping:
                 r += '{'
-            for idx, (k, v) in enumerate(input_dict.items()):
+            for idx, (k, v) in enumerate(
+                    sorted(input_dict.items(), key=lambda x: str(x))):
```

Comment:
```suggestion
                    sorted(input_dict.items(), key=lambda x: str(x[0]))):
```

Reply:
It may be enough to only sort it by the key (`x[0]`).

Reply:
I don't think so. 

configuration file's key in other project like mmdetection has the following naming format. 

```bash
...
train_pipeline = ...
train_dataloader = ....
val_pipeline = ...
val_dataloader = ....
val_evaluator = ... 
...
```
So 1 character won't be enough.
I think 4 characters ("train") is a better value.

Reply:
x[0] means the first item (corresponding to key) rather than the first char of key.

Reply:
Oh i got it now.  so sleepy in the morning

Reply:
Thanks for your contribution.